sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -44,12 +48,11 @@ def _csr_row_norms(
 
     cdef floating[::1] norms = np.zeros(n_samples, dtype=dtype)
```

Comment:
looking at the cythonization error, you forgot to declare ``squared_row_norms``. You forgot this line in your renaming
```suggestion
    cdef floating[::1] squared_row_norms = np.zeros(n_samples, dtype=dtype)
```